self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryProjectsTestCase.java

```diff
@@ -0,0 +1,97 @@
+package com.selfxdsd.core.mock;
+
+import com.selfxdsd.api.*;
+import com.selfxdsd.api.storage.Storage;
+import com.selfxdsd.core.projects.UserProjects;
+import org.junit.Test;
+import java.util.ArrayList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for class {@link InMemoryProjects}.
+ * @author hpetrila
+ * @version $Id$
+ * @since 0.0.1
+ * @todo #16:30min Continue writing unit tests for the InMemory storage
+ *  infrastructure. In the end all classes and methods should be covered
+ *  with unit tests so we can rely 100% on them.
+ */
+public final class InMemoryProjectsTestCase {
+
+    /**
+     * Register a project.
+     */
+    @Test
+    public void projectRegister() {
+        final Storage storage = new InMemory();
+        ProjectManager projectManager = storage
+                .projectManagers().pick();
+        InMemoryProjects projects = (InMemoryProjects)
+                storage.projects();
+        Project project = projects.register(
+                mock(Repo.class), projectManager);
+
+        assertThat(project.projectManager(),
+                equalTo(projectManager));
```

Comment:
@HoreaPetrila mai bine foloseste ``is(...)`` pentru ca ar trebui sa fie exact aceeasi referinta.